cel/canonical_eventlog.go

```diff
@@ -340,3 +348,39 @@ func DecodeToCELR(buf *bytes.Buffer) (r Record, err error) {
 	}
 	return r, nil
 }
+
+// Replay takes the digests from a Canonical Event Log and carries out the
+// extend sequence for each PCR in the log. It then compares the final digests
+// against a bank of PCR values to see if they match.
+func (c *CEL) Replay(bank *pb.PCRs) error {
```

Comment:
I am wondering what's the behavior of Replay.

* I am thinking should `Replay()` just return a list of all final PCRs values involved in the CEL, and we do the validation in later/in another function

Reply:
I'm  what the point of replay is without comparing it. What's your underlying concern? I can rename it `Verify` if you don't like the name and pull replay out to an unexported function if you want to use it for something else.

Reply:
My main concern is the hash algorithm selection.
When a user call Replay(), they won't have an idea about the what pcr hashalgo they are facing (they can try to read the Record directly to get the hashalgo but it won't be straight forward), they might be confused about what hashalgo to set in *pb.PCRs.

Reply:
talked offline:
We will unexport Replay for now, since there is no need for this to be user facing at the moment. It will be integrated into VerifyAttestation or ParseMachineState where the caller will try all the given banks to replay. If the hashAlgo is not found in even one record, the whole replay will fail.

Also, we discussed how this replay behavior differs from go-attestation. go-attestation checks that the intersecting PCRs of the log and the bank match digests. Our approach will be to see if PCRs seen in the CEL are a subset of the ones passed in. If so, we check that the subset's PCRs match digests with the passed in bank.